stacks_queues/queue.py

```diff
@@ -15,48 +15,98 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == self.buffer_size:
+            raise QueueFullException("Queue if full")
+
+        # condition for empty queue
+        elif (self.front == -1):
+            self.front = 0
+            self.rear = 0
+            self.store[self.rear] = element
+            self.size +=1 
+        
+        else:
+            # next position of rear
+            self.rear = (self.rear + 1) % self.buffer_size
+            self.store[self.rear] = element
+            self.size +=1
 
     def dequeue(self):
         """ Removes an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
             returns None
         """
-        pass
+        if self.size == 0:
+            raise QueueEmptyException("Queue is empty")
+
+        data = self.store[self.front]
+        self.store[self.front] = None
+        self.size -= 1
+
+        if self.front == self.rear:
+            self.front = -1
+            self.rear = -1
+        elif self.front == self.buffer_size-1:
+            self.front = 0
+        else:
+            self.front +=1
+
+        return data
```

Comment:
👍 However this could be simplified with the `self.front = (self.front + 1)  % self.buffer_size